### PR TITLE
Adds support for D1 

### PR DESCRIPTION
**Pull request by Cyb3r-Jak3**
Closes https://github.com/cloudflare/terraform-provider-cloudflare/issues/2544

Adds support for D1 databases

~~Also, a small docs update clarifying that logging options are only valid for the skip action. There is an API error of `logging options only allowed in the skip action (20018)` that is returned if you try with other rules. Might be better to add a validator for it but not 100% on how to do so.~~

### CODE REVIEW

**Comment by Cyb3r-Jak3** [comment]
```bash
$ TF_ACC=1 go test ./internal/framework/service/d1/ -v -count 1 -run ^TestAccCloudflareD1Database_ -timeout 120m -parallel 1
=== RUN   TestAccCloudflareD1Database_Basic
=== PAUSE TestAccCloudflareD1Database_Basic
=== CONT  TestAccCloudflareD1Database_Basic
--- PASS: TestAccCloudflareD1Database_Basic (3.57s)
PASS
ok      github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/d1   
```

**Comment by jacobbednarz** [comment]
> Also, a small docs update clarifying that logging options are only valid for the skip action. There is an API error of logging options only allowed in the skip action (20018) that is returned if you try with other rules. Might be better to add a validator for it but not 100% on how to do so.

can you please split this into two PRs? I don't like unrelated changes being merged in a single PR.

**Comment by Cyb3r-Jak3** [comment]
Done

File: .changelog/2850.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:new-resource
+resource/d1: Adds support for D1 databases
```
Comment (jacobbednarz) [inline .changelog/2850.txt:2]:
```suggestion
cloudflare_d1_database
```

**Comment by jacobbednarz** [comment]
thanking you!

acceptance tests all looking good

```
TF_ACC=1 go test ./internal/framework/service/d1 -v -run "^TestAccCloudflareD1Database_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareD1Database_Basic
=== PAUSE TestAccCloudflareD1Database_Basic
=== CONT  TestAccCloudflareD1Database_Basic
--- PASS: TestAccCloudflareD1Database_Basic (13.86s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/framework/service/d1	14.775s
```

**Pull request merged**